spacy/pipeline/trainable_pipe.pyx

```diff
@@ -169,6 +219,17 @@ cdef class TrainablePipe(Pipe):
         """
         raise NotImplementedError(Errors.E931.format(parent="TrainablePipe", method="get_loss", name=self.name))
 
+    def get_teacher_student_loss(self, teacher_scores, student_scores):
+        """Calculate the loss and its gradient for a batch of student
+        scores, relative to teacher scores.
+
+        teacher_scores: Scores representing the teacher model's predictions.
+        student_scores: Scores representing the student model's predictions.
+
+        DOCS: https://spacy.io/api/pipe#get_teacher_student_loss
+        """
+        raise NotImplementedError(Errors.E931.format(parent="TrainablePipe", method="get_teacher_student_loss", name=self.name))
```

Comment:
It would be nice to have a mechanism that can tell you whether distillation is in fact implemented for a given pipe, something like `is_distillable()`. As we have a default dummy implementation here of `get_teacher_student_loss` we can't implement it checking whether the right functions are available though, so maybe it should be `False` by default and pipes should set it specifically to `True` when they've implemented the behaviour fully?

Reply:
I am a bit worried that downstream pipes forget to set this. Though, I think it would be enough to check in `is_distillable()` whether `distill` and `get_student_teacher_loss` are still the base class (`TrainablePipe`) methods? Then `is_distillable()` would return `True` if a pipe implements either of these methods. Let me implement it in this way and see if you like it...

Reply:
Added.